pkg/controller/cloneset/cloneset_controller_test.go

```diff
@@ -131,6 +137,109 @@ func TestReconcile(t *testing.T) {
 	testUpdate(g, instance)
 }
 
+func TestClaimPods(t *testing.T) {
```

Comment:
Can you simply test the `claimPods` method functionality without setup the TestManager? The test can be as simple as just passing a cloneset object and an array of Pods. You don't need a live apiserver to create real objects.

Reply:
use fakeClient instead, done.

Reply:
I  am actually suggesting to remove entire gomega code and client code, even removing the input parameter t. 
Will that work?

Reply:
I think client code can't be moved, it will be used in the interface of k8s update.
can you show me some example?

Reply:
line 191, Will change `claimed, err := reconciler.claimPods(instance, test.pods)` to `claimed, err := claimPods(instance, test.pods)` work?

Reply:
Yes. you are right.  I overlook that aspect. sorry.